src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -72,16 +74,61 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
             [this](const std::vector<char>& message)
             {
                 std::cout << "DatabaseFeedManager::update" << std::endl;
-                auto eventContext = std::make_shared<EventContext>(
-                    EventContext {.indexerConnector = m_indexerConnector, .message = message});
+                auto eventContext = std::make_shared<EventContext>(EventContext {.message = message});
                 auto eventDecoder = std::make_shared<EventDecoder>();
                 eventDecoder->setLast(std::make_shared<StoreModel>());
-                eventDecoder->setLast(std::make_shared<FeedIndexer>());
+                eventDecoder->setLast(std::make_shared<FeedIndexer<TIndexerConnector>>(m_indexerConnector));
                 eventDecoder->setLast(std::make_shared<ScanDispatcher>());
 
                 eventDecoder->handleRequest(eventContext);
             });
     }
+    /**
+     * @brief Get the Vulnerabilities Candidates information.
+     *
+     * @param cnaName RocksDB table identifier.
+     * @param context Scan context structure.
+     * @param callback Store vulnerability data.
+     */
+    void getVulnerabilitiesCandidates(const std::string& cnaName,
```

Comment:
What is the error flow of this function? If an exception is thrown, how notify which error causes the caller?

Reply:
I tried to respect as much as possible the definition mentioned in the issue. I think the best way to do this is by logging the errors, warnings, and info messages. Throwing exceptions in that function will break the loop.

Reply:
What do think about refactoring this function in this way:

```cpp
        try
        {
            Utils::RocksDBWrapper rocksDBWrapper(ROCKS_DB_PATH);
            if (!context.contextData.empty())
            {
                try
                {
                    packageName = context.contextData.at("data").at("name");
                    for (const auto& [key, value] : rocksDBWrapper.seek(packageName))
                    {
                        if (callback(reinterpret_cast<const char*>(value.data()), value.size()))
                        {
                            // Logging
                        }
                    }
                }
                catch (...)
                {
                        //Loggin
                        std::cerr<<"Invalid package\n";
                }
            }
        }
        catch (...)
        {
                        //Loggin
                        std::cerr<<"Unable to open database\n";
        }
    }
```